packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -109,7 +116,105 @@ export default createRule<
     const seenLogicals = new Set<TSESTree.LogicalExpression>();
 
     return {
-      // specific handling for `(foo ?? {}).bar` / `(foo || {}).bar`
+      // specific handling for `if (foo) { foo.bar(); }` / `if (foo) foo.bar();`
+      'IfStatement[consequent.type=BlockStatement][consequent.body.length=1], IfStatement[consequent.type=ExpressionStatement]':
+        (
+          node: {
+            consequent: {
+              type:
+                | AST_NODE_TYPES.BlockStatement
+                | AST_NODE_TYPES.ExpressionStatement;
+            };
+          } & TSESTree.IfStatement,
```

Comment:
[Style] Nit: as nice as these types are, there's no need for the fancy `consequent` specificity. Removing it doesn't cause any new type errors.

```suggestion
          node: TSESTree.IfStatement,
```